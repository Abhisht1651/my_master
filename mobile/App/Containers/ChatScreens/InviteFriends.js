// @flow
import React from 'react'
import { NavigationContainerProps } from '@react-navigation/native'
import { View, Text, TouchableOpacity, ScrollView, Image } from 'react-native'
import ListOfUsers from '../../Components/ListOfUsers/ListOfUsers'
import Images from '../../Themes/Images'

// Styles
import styles from './ScreenStyles/InviteFriendsStyles'

type Props = {} & NavigationContainerProps

const InviteFriends = (props: Props) => {
  const { navigation } = props
  return (
    <View style={styles.screen}>
      <View style={styles.topBox}>
        <View style={styles.backbutton}>
          <TouchableOpacity onPress={() => navigation.navigate('MessageScreen')}>
            <Image source={Images.arrowleft} style={styles.arrowleft} />
          </TouchableOpacity>
        </View>
        <Text style={styles.titleFont}>Invite Friends</Text>
        <TouchableOpacity>
          <Text style={styles.importFont}>Import</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.mainBox}>
        <View style={styles.textBox}>
          <Text style={styles.textFont}>Based on your contacts</Text>
        </View>
        <View style={styles.listBox}>
          <ScrollView>
            <ListOfUsers
              name="Laura Leucona"
              optionalText="laura@newzera.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
              image={Images.profile2}
            />
            <ListOfUsers
              name="Sachin Tendulkar"
              optionalText="notout100@gmail.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
              image={Images.profile1}
            />
            <ListOfUsers
              name="Brian Lara"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
              image={Images.profilePicture3}
            />
            <ListOfUsers
              name="Sourav Ganguly"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
              image={Images.profilePicture4}
            />
            <ListOfUsers
              name="Laura Leucona"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
            />
            <ListOfUsers
              name="Laura Leucona"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
            />
            <ListOfUsers
              name="Laura Leucona"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
            />
            <ListOfUsers
              name="Laura Leucona"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
            />
            <ListOfUsers
              name="Laura Leucona"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
            />
            <ListOfUsers
              name="Laura Leucona"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
            />
            <ListOfUsers
              name="Laura Leucona"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
            />
            <ListOfUsers
              name="Laura Leucona"
              optionalText="email@email.com"
              buttonText="Invite"
              buttonTextAfterPress="Invited"
            />
          </ScrollView>
        </View>
      </View>
    </View>
  )
}

export default InviteFriends
