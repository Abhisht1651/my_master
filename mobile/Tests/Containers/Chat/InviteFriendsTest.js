import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, Text, Image, ScrollView } from 'react-native'
import InviteFriends from '../../../App/Containers/ChatScreens/InviteFriends'

describe('InviteFriends', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<InviteFriends />) // no compile-time error
  })
  it('InviteFriends component renders correctly', () => {
    const wrapper = shallow(<InviteFriends />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 2 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(2)
  })
  it('should render 3 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(3)
  })
  it('should render 1 Image', () => {
    expect(wrapper.find(Image)).toHaveLength(1)
  })
  it('should render 1 ScrollView', () => {
    expect(wrapper.find(ScrollView)).toHaveLength(1)
  })
})
