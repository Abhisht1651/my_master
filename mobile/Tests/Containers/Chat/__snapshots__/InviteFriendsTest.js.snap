// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteFriends InviteFriends component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": "column",
      "height": 1334,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#E6ECED",
        "borderBottomWidth": 2,
        "elevation": 2,
        "flexDirection": "row",
        "height": 110.07142857142857,
        "justifyContent": "space-between",
        "paddingHorizontal": 48,
        "width": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 34,
              "width": 44,
            }
          }
        />
      </ForwardRef>
    </View>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 40,
          "letterSpacing": -0.28,
        }
      }
    >
      Invite Friends
    </Text>
    <ForwardRef>
      <Text
        style={
          Object {
            "color": "#7B46E4",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 32,
            "letterSpacing": -0.18,
          }
        }
      >
        Import
      </Text>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "height": 1334,
        "paddingHorizontal": 44,
        "paddingTop": 55.857142857142854,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "height": 39.42857142857142,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Medium",
            "fontSize": 36,
            "letterSpacing": -0.24,
          }
        }
      >
        Based on your contacts
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingTop": 42.714285714285715,
          "width": "100%",
        }
      }
    >
      <ScrollViewMock>
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="laura@newzera.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Sachin Tendulkar"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="notout100@gmail.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Brian Lara"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Sourav Ganguly"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
        <ListOfUsers
          buttonText="Invite"
          buttonTextAfterPress="Invited"
          image={
            Object {
              "default": "",
            }
          }
          name="Laura Leucona"
          needoptiontext={false}
          onCancelPress={[Function]}
          onPress={[Function]}
          optionalText="email@email.com"
          profileType={1}
          type={1}
        />
      </ScrollViewMock>
    </View>
  </View>
</View>
`;
